### DISCUSSION TITLE
Choice between API Gateway

### DISCUSSION THREAD

**Discussion by sycured** [supabase/supabase#2876] [Feature Requests]
# Feature request

Let the user decide which API Gateway he wants

## Describe the solution you'd like

The possibility to the user during the setup to choose the API Gateway

## Describe alternatives you've considered

Exporting Kong configuration and implement it directly in Nginx with/without using Consul or also Tyk

**Comment by darora** [comment]
@sycured is there some particular functionality you're looking for that Kong doesn't accommodate?

**Comment by sycured** [comment]
@darora I've a custom Nginx with some custom modules (C/Lua/Rust) so it's like accepting to break years to have it running well.

Another thing is about the scaling and routing/permissions, I've a very close integration between Nginx and Consul to avoid any reload/reboot when a change is coming but also using Consul with it's Intention (https://www.consul.io/docs/connect/intentions) to avoid a lot of bad practices and having a clear ACL matrix.

Tyk can be a good challenger: https://www.google.com/amp/s/www.bbva.com/en/api-gateways-kong-vs-tyk/amp/

And this is another good alternative: https://github.com/apache/apisix

**Reply by juzhiyuan** [reply]
Yes, and Apache APISIX has released the first LTS version 😄 see https://apisix.apache.org/ and https://github.com/apache/apisix/discussions/5156

**Comment by kiwicopple** [comment]
Hi @sycured - for now I'm going to convert this to a Discussion. It's worth chatting about but we're all-in on Kong right now and we want to keep the core team focused. Hopefully the community can help with this and then we can come up with a few Gateway variations in the self hosting :)

**Reply by moonming** [reply]
If use Apache APISIX instead of kong, then you don’t need to maintain Postgres and more cloud-native

**Reply by sycured** [reply]
> If use Apache APISIX instead of kong, then you don’t need to maintain Postgres and more cloud-native

APISIX is interesting but I'll need to look if I can replace etcd requirement by consul easily or not due to my usage of Nomad

**Reply by NiklasPor** [reply]
@sycured sorry to revive this 3 years later, but did you have working version of Supabase deployable with Nomad? I already spent a vew hours trying to convert the compose file to Nomad. Would love to know!
